spyrmsd/graph.py

```diff
@@ -1,42 +1,97 @@
+from spyrmsd import constants
+
+import numpy as np
+import os
+
+available_backends = []
+
 try:
     from spyrmsd.graphs.gt import (
-        cycle,
-        graph_from_adjacency_matrix,
-        lattice,
-        match_graphs,
-        num_edges,
-        num_vertices,
-        vertex_property,
+        cycle as gt_cycle,
+        graph_from_adjacency_matrix as gt_graph_from_adjacency_matrix,
+        lattice as gt_lattice,
+        match_graphs as gt_match_graphs,
+        num_edges as gt_num_edges,
+        num_vertices as gt_num_vertices,
+        vertex_property as gt_vertex_property,
     )
-
+    available_backends.append("graph-tool")
 except ImportError:
-    try:
-        from spyrmsd.graphs.nx import (
-            cycle,
-            graph_from_adjacency_matrix,
-            lattice,
-            match_graphs,
-            num_edges,
-            num_vertices,
-            vertex_property,
-        )
-    except ImportError:
-        raise ImportError("graph_tool or NetworkX libraries not found.")
-
-__all__ = [
-    "graph_from_adjacency_matrix",
-    "match_graphs",
-    "vertex_property",
-    "num_vertices",
-    "num_edges",
-    "lattice",
-    "cycle",
-    "adjacency_matrix_from_atomic_coordinates",
-]
-
-import numpy as np
-
-from spyrmsd import constants
+    pass
+    
+try:
+   from spyrmsd.graphs.nx import (
+        cycle as nx_cycle,
+        graph_from_adjacency_matrix as nx_graph_from_adjacency_matrix,
+        lattice as nx_lattice,
+        match_graphs as nx_match_graphs,
+        num_edges as nx_num_edges,
+        num_vertices as nx_num_vertices,
+        vertex_property as nx_vertex_property,
+    )
+   available_backends.append("networkx")
+except ImportError:
+   pass
+        
+def get_available_backends():
+    return available_backends 
+
+def set_backend(backend):
+
+    ## Get the current backend
+    current_backend = str(os.environ.get("SPYRMSD_BACKEND"))
+  
+    ## Check if the backend is valid
+    if backend.lower() in ["graph_tool", "graphtool", "graph-tool", "graph tool", "gt"]:
+        backend = "graph-tool"
+    elif backend.lower() in ["networkx", "nx"]:
+        backend = "networkx"
+    else:
+        raise ValueError("This backend is not recognized or supported")
```

Comment:
I think this should be its own function. I also think that it would be better to move the list of possible values in their own modules (`nx.py` and `gt.py`, respectively).

Reply:
I agree it would be cleaner to have some kind of variable which can store the aliases. However, I think in the current implementation it would give problems if we move it to the backend file (gt.py or nx.py)? Let's say graph-tool isn't installed, wouldn't trying to get the aliases from gt.py result in import errors (because gt.py imports graph-tool). Or is there an easy work around for this?

Reply:
Right, I didn't think of that! In this case, maybe it's still worth putting the mapping into a (private) dictionary at the top of this module, and have a `_validate_backend` function taking `backend` as input and returning the standard backend as output? I think the check below can be also incorporated in such function  (since the backend needs to be available).